Resources/config/imagine.xml

```diff
@@ -32,6 +32,39 @@
 
     <services>
 
+        <!-- Factory services -->
+        <service id="liip_imagine.factory.config.filter_set" class="Liip\ImagineBundle\Factory\Config\FilterSetFactory" />
+        <service id="Liip\ImagineBundle\Factory\Config\FilterSetFactory" alias="liip_imagine.factory.config.filter_set"/>
+
+        <service id="liip_imagine.factory.config.filter.crop" class="Liip\ImagineBundle\Factory\Config\Filter\CropFactory" />
+        <service id="Liip\ImagineBundle\Factory\Config\Filter\CropFactory" alias="liip_imagine.factory.config.filter.crop"/>
+
+        <service id="liip_imagine.factory.config.filter.strip" class="Liip\ImagineBundle\Factory\Config\Filter\StripFactory" />
+        <service id="Liip\ImagineBundle\Factory\Config\Filter\StripFactory" alias="liip_imagine.factory.config.filter.strip"/>
+
+        <service id="liip_imagine.factory.config.filter.thumbnail" class="Liip\ImagineBundle\Factory\Config\Filter\ThumbnailFactory" />
+        <service id="Liip\ImagineBundle\Factory\Config\Filter\ThumbnailFactory" alias="liip_imagine.factory.config.filter.thumbnail"/>
+
+        <!-- Config services -->
+        <service id="liip_imagine.config.filter_factory_collection" class="Liip\ImagineBundle\Config\FilterFactoryCollection">
+            <argument type="service" id="liip_imagine.factory.config.filter.crop"/>
+            <argument type="service" id="liip_imagine.factory.config.filter.strip"/>
+            <argument type="service" id="liip_imagine.factory.config.filter.thumbnail"/>
+        </service>
+        <service id="Liip\ImagineBundle\Config\FilterFactoryCollection" alias="liip_imagine.config.filter_factory_collection"/>
```

Comment:
neither do we need autowiring for this. the only thing that should be available for autowiring should be the actual configuration object.

Reply:
done

Reply:
@dbu, btw, do we need to have SF4 compatibility ?

Reply:
it would be nice, but i would not mix it into this pull request.

if we have new services that should be autowire-enabled, we can already define the right alias here, but i would not do anything outside the scope of modeling the configuration.